lib/puppet/pops/types/types.rb

```diff
@@ -701,6 +701,23 @@ def each(&block)
     block_given? ? r.each(&block) : r
   end
 
+  def generalize
+    # General form of an Enum is a String
+    PStringType::DEFAULT
+  end
+
+  def normalize(guard = nil)
+    # General form of an Enum is a String
+    case @values.size
+    when 0
+      PStringType::DEFAULT
+    when 1
+      PStringType.new(@values[0])
+    else
+      self
+    end
+  end
```

Comment:
There is no normalization on a String type. It simply returns itself.

There didn't use to be any normalization on the Enum type either. I thought it made sense to normalize a one-element Enum to a String since there's not much point in having it as a Enum unless it has multiple choices. I can remove it for sure.